### PR TITLE
[FLINK-30429][client] Fix IllegalArgumentException when no argument in flink executable

### PR DESCRIPTION
**Pull request by gaborgsomogyi**
## What is the purpose of the change

Flink now throws exception in command line tools:
```
$ flink
...
java.lang.IllegalArgumentException: 1 > 0
	at java.base/java.util.Arrays.copyOfRange(Arrays.java:3990)
	at java.base/java.util.Arrays.copyOfRange(Arrays.java:3950)
	at org.apache.flink.client.cli.CliFrontend.main(CliFrontend.java:1170)
```
In this PR I've fixed it.

## Brief change log

Fixed the exception.

## Verifying this change

* New unit test
* Manually, build Flink and start `flink` command. The tool must print usage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by gaborgsomogyi** [comment]
cc @gyfora

File: flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java [hunk 1]
```diff
@@ -1149,6 +1150,11 @@ public int parseAndRun(String[] args) {
 
     /** Submits the job based on the arguments. */
     public static void main(final String[] args) {
+        System.exit(mainInternal(args));
```
Comment (gyfora) [inline flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java:1153]: We need to change this to
```
 int retCode = 31;
 try {
 retCode = mainInternal();
} finallly{System.exit(retCode)}

to preserve the original behavior I think.

**Comment by gaborgsomogyi** [comment]
@flinkbot run azure

**Comment by gaborgsomogyi** [comment]
@flinkbot run azure

**Comment by gaborgsomogyi** [comment]
@flinkbot run azure

File: flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java (hunk 1 above)
Comment (gaborgsomogyi) [reply flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java]: Added.

**Comment by gyfora** [comment]
@flinkbot run azure

**Comment by gyfora** [comment]
@flinkbot run azure

**Pull request merged**